### PR TITLE
Feature/risky get ball

### PR DESCRIPTION
**Pull request by timovdk**
Created a risky get ball, which basically gets the ball and prepares three offensive receive positions, so if the next play is a pass, we already have three offensive pass options (at least that's the theory).
closesl #1111

### CODE REVIEW

File: src/stp/new_plays/GetBallRisky.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+GetBallRisky::GetBallRisky() : Play() {
+    startPlayInvariants.clear();
+    startPlayInvariants.emplace_back(std::make_unique<invariant::NormalPlayGameStateInvariant>());
+    startPlayInvariants.emplace_back(std::make_unique<invariant::BallIsFreeInvariant>());
+    //startPlayInvariants.emplace_back(std::make_unique<invariant::WeHaveMajorityInvariant>());
+
+    keepPlayInvariants.clear();
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::NormalPlayGameStateInvariant>());
+    keepPlayInvariants.emplace_back(std::make_unique<invariant::BallIsFreeInvariant>());
+    //keepPlayInvariants.emplace_back(std::make_unique<invariant::WeHaveMajorityInvariant>());
```
Comment (timovdk) [inline src/stp/new_plays/GetBallRisky.cpp]:
```suggestion
    startPlayInvariants.emplace_back(std::make_unique<invariant::WeHaveMajorityInvariant>());

    keepPlayInvariants.clear();
    keepPlayInvariants.emplace_back(std::make_unique<invariant::NormalPlayGameStateInvariant>());
    keepPlayInvariants.emplace_back(std::make_unique<invariant::BallIsFreeInvariant>());
    keepPlayInvariants.emplace_back(std::make_unique<invariant::WeHaveMajorityInvariant>());
```

File: src/stp/new_plays/GetBallRisky.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    enemyRobots.erase(std::remove_if(enemyRobots.begin(), enemyRobots.end(), [&](const auto enemyRobot) -> bool { return enemyRobot->getId() == enemyAttacker->getId(); }));
+
+    auto enemyClosestToGoal = world->getWorld()->getRobotClosestToPoint(field.getOurGoalCenter(), enemyRobots);
+
+    stpInfos["keeper"].setEnemyRobot(world->getWorld()->getRobotClosestToBall(world_new::them));
+
+    // TODO: determine better future receive positions
+    stpInfos["receiver_0"].setPositionToMoveTo(Vector2(field.getFieldLength() / 4, field.getFieldWidth() / 4));
+    stpInfos["receiver_1"].setPositionToMoveTo(Vector2(field.getFieldLength() / 4, -field.getFieldWidth() / 4));
+    stpInfos["receiver_2"].setPositionToMoveTo(Vector2(field.getFieldLength() / 4, 0));
```
Comment (timovdk) [inline src/stp/new_plays/GetBallRisky.cpp:58]: Can I get input on how to determine good pass positions without pagmo? probably needs to share the algorithm with the actual pass play @RobotJesse @jordiluong

File: src/stp/new_plays/GetBallRisky.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+bool GetBallRisky::shouldRoleSkipEndTactic() { return false; }
+
+Dealer::FlagMap GetBallRisky::decideRoleFlags() const noexcept {
+    Dealer::FlagMap flagMap;
+    Dealer::DealerFlag keeper(DealerFlagTitle::KEEPER, DealerFlagPriority::KEEPER);
+    Dealer::DealerFlag ball_getter(DealerFlagTitle::CLOSE_TO_BALL, DealerFlagPriority::REQUIRED);
+    Dealer::DealerFlag not_important(DealerFlagTitle::ROBOT_TYPE_50W, DealerFlagPriority::LOW_PRIORITY);
+    Dealer::DealerFlag closeToBallFlag(DealerFlagTitle::CLOSE_TO_BALL, DealerFlagPriority::HIGH_PRIORITY);
```
Comment (timovdk) [inline src/stp/new_plays/GetBallRisky.cpp]:
```suggestion
```

File: src/stp/new_plays/GetBallRisky.cpp (hunk 2 above)
Comment (RobotJente) [reply src/stp/new_plays/GetBallRisky.cpp:58]: Yeah, I'm trying to find a good algorithm. I think once the new pass play (without pagmo) is done we will have a bit more information about this

File: src/stp/new_plays/GetBallRisky.cpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                                                                                 std::make_unique<role::Defender>(role::Defender("midfielder_2"))};
+}
+
+uint8_t GetBallRisky::score(world_new::World* world) noexcept { return 120; }
+
+void GetBallRisky::calculateInfoForRoles() noexcept {
+    auto enemyRobots = world->getWorld()->getThem();
+    auto enemyAttacker = world->getWorld()->getRobotClosestToBall(world_new::them);
+
+    enemyRobots.erase(std::remove_if(enemyRobots.begin(), enemyRobots.end(), [&](const auto enemyRobot) -> bool { return enemyRobot->getId() == enemyAttacker->getId(); }));
```
Comment (RobotJente) [inline src/stp/new_plays/GetBallRisky.cpp:49]: This is a copy of getThem(), right? Otherwise we are killing robots here... :smile:

Comment (timovdk) [reply src/stp/new_plays/GetBallRisky.cpp:49]: it is a const reference to the non-owning container of RobotViews of the enemy team, and the world where we get that from should be updated every tick.
EDIT: This was not the case, so made an issue #1119

**Review by RobotJente — APPROVED** [review]

**Pull request merged**